sympy/core/tests/test_evalf.py

```diff
@@ -663,6 +663,14 @@ def test_evalf_with_zoo():
     assert x.evalf(subs={x: zoo}) == zoo
 
 
+def test_issue_20733():
+    expr = 1/((x - 9)*(x - 8)*(x - 7)*(x - 4)**2*(x - 3)**3*(x - 2))
+    assert str(expr.evalf(1, subs={x:1})) == '-4.e-5'
+    assert str(expr.evalf(2, subs={x:1})) == '-4.1e-5'
+    assert str(expr.evalf(11, subs={x:1})) == '-4.1335978836e-5'
+    assert str(expr.evalf(20, subs={x:1})) == '-0.000041335978835978835979'
```

Comment:
It would be better I think to test an expression that generates the issue more reliably. 
```python
In [86]: expr = Mul(*((x - i) for i in range(2, 1000)))

In [87]: [expr.evalf(n, subs={x: 1}).evalf(2) for n in range(10, 100)]
Out[87]: 
[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
```
The point is having many args in the Mul so we can hit that if condition many times. This expression fails for anything less than 500 digits of precision.

Reply:
Yeah makes sense .I had added issue specific tests which still could be maintained .

Reply:
We can have both kinds of tests

Reply:
This PR is good to go if you add the test I showed above.

Reply:
Yes ,I will start working on it .I would like some guidance on what is the standard in SymPy when dealing with a test having large number of digits involved as I have never encountered any till now .
Like the result of the above suggested test is shown below and it is as expected .But obviously it would not be right to assert and compare this whole list .I guess such tests are handled by executing the command but asserting for only few random indexes of the list . My diff is ready and I'll commit it as soon as this is confirmed .Thanks !

`[4.027900501e+2561, 4.0279005013e+2561, 4.02790050127e+2561, 4.027900501272e+2561, 4.0279005012722e+2561, 4.02790050127221e+2561, 4.027900501272210e+2561, 4.0279005012722099e+2561, 4.02790050127220995e+2561, 4.027900501272209945e+2561, 4.0279005012722099454e+2561, 4.02790050127220994538e+2561, 4.027900501272209945382e+2561, 4.0279005012722099453824e+2561, 4.02790050127220994538241e+2561, 4.027900501272209945382407e+2561, 4.0279005012722099453824067e+2561, 4.02790050127220994538240675e+2561, 4.027900501272209945382406746e+2561, 4.0279005012722099453824067460e+2561, 4.02790050127220994538240674598e+2561, 4.027900501272209945382406745976e+2561, 4.0279005012722099453824067459760e+2561, 4.02790050127220994538240674597602e+2561, 4.027900501272209945382406745976016e+2561, 4.0279005012722099453824067459760159e+2561, 4.02790050127220994538240674597601587e+2561, 4.027900501272209945382406745976015873e+2561, 4.0279005012722099453824067459760158731e+2561, 4.02790050127220994538240674597601587307e+2561, 4.027900501272209945382406745976015873067e+2561, 4.0279005012722099453824067459760158730668e+2561, 4.02790050127220994538240674597601587306682e+2561, 4.027900501272209945382406745976015873066815e+2561, 4.0279005012722099453824067459760158730668155e+2561, 4.02790050127220994538240674597601587306681546e+2561, 4.027900501272209945382406745976015873066815458e+2561, 4.0279005012722099453824067459760158730668154576e+2561, 4.02790050127220994538240674597601587306681545756e+2561, 4.027900501272209945382406745976015873066815457565e+2561, 4.0279005012722099453824067459760158730668154575647e+2561, 4.02790050127220994538240674597601587306681545756471e+2561, 4.027900501272209945382406745976015873066815457564711e+2561, 4.0279005012722099453824067459760158730668154575647110e+2561, 4.02790050127220994538240674597601587306681545756471104e+2561, 4.027900501272209945382406745976015873066815457564711036e+2561, 4.0279005012722099453824067459760158730668154575647110365e+2561, 4.02790050127220994538240674597601587306681545756471103647e+2561, 4.027900501272209945382406745976015873066815457564711036474e+2561, 4.0279005012722099453824067459760158730668154575647110364745e+2561, 4.02790050127220994538240674597601587306681545756471103647447e+2561, 4.027900501272209945382406745976015873066815457564711036474474e+2561, 4.0279005012722099453824067459760158730668154575647110364744736e+2561, 4.02790050127220994538240674597601587306681545756471103647447358e+2561, 4.027900501272209945382406745976015873066815457564711036474473578e+2561, 4.0279005012722099453824067459760158730668154575647110364744735779e+2561, 4.02790050127220994538240674597601587306681545756471103647447357788e+2561, 4.027900501272209945382406745976015873066815457564711036474473577877e+2561, 4.0279005012722099453824067459760158730668154575647110364744735778773e+2561, 4.02790050127220994538240674597601587306681545756471103647447357787726e+2561, 4.027900501272209945382406745976015873066815457564711036474473577877262e+2561, 4.0279005012722099453824067459760158730668154575647110364744735778772624e+2561, 4.02790050127220994538240674597601587306681545756471103647447357787726239e+2561, 4.027900501272209945382406745976015873066815457564711036474473577877262386e+2561, 4.0279005012722099453824067459760158730668154575647110364744735778772623864e+2561, 4.02790050127220994538240674597601587306681545756471103647447357787726238637e+2561, 4.027900501272209945382406745976015873066815457564711036474473577877262386373e+2561, 4.0279005012722099453824067459760158730668154575647110364744735778772623863727e+2561, 4.02790050127220994538240674597601587306681545756471103647447357787726238637266e+2561, 4.027900501272209945382406745976015873066815457564711036474473577877262386372663e+2561, 4.0279005012722099453824067459760158730668154575647110364744735778772623863726629e+2561, 4.02790050127220994538240674597601587306681545756471103647447357787726238637266287e+2561, 4.027900501272209945382406745976015873066815457564711036474473577877262386372662869e+2561, 4.0279005012722099453824067459760158730668154575647110364744735778772623863726628688e+2561, 4.02790050127220994538240674597601587306681545756471103647447357787726238637266286879e+2561, 4.027900501272209945382406745976015873066815457564711036474473577877262386372662868789e+2561, 4.0279005012722099453824067459760158730668154575647110364744735778772623863726628687892e+2561, 4.02790050127220994538240674597601587306681545756471103647447357787726238637266286878923e+2561, 4.027900501272209945382406745976015873066815457564711036474473577877262386372662868789231e+2561, 4.0279005012722099453824067459760158730668154575647110364744735778772623863726628687892313e+2561, 4.02790050127220994538240674597601587306681545756471103647447357787726238637266286878923132e+2561, 4.027900501272209945382406745976015873066815457564711036474473577877262386372662868789231316e+2561, 4.0279005012722099453824067459760158730668154575647110364744735778772623863726628687892313162e+2561, 4.02790050127220994538240674597601587306681545756471103647447357787726238637266286878923131619e+2561, 4.027900501272209945382406745976015873066815457564711036474473577877262386372662868789231316186e+2561, 4.0279005012722099453824067459760158730668154575647110364744735778772623863726628687892313161859e+2561, 4.02790050127220994538240674597601587306681545756471103647447357787726238637266286878923131618588e+2561, 4.027900501272209945382406745976015873066815457564711036474473577877262386372662868789231316185880e+2561, 4.0279005012722099453824067459760158730668154575647110364744735778772623863726628687892313161858799e+2561, 4.02790050127220994538240674597601587306681545756471103647447357787726238637266286878923131618587993e+2561]`

Reply:
The diff is as follows -
```
+    expr = Mul(*((x - i) for i in range(2, 1000)))
+    result = [expr.evalf(n, subs={x: 1}) for n in range(10, 100)]
+    assert str(result[0]) == '4.027900501e+2561'
+    assert str(result[10]) == '4.0279005012722099454e+2561'
+    assert str(result[50]) == '4.02790050127220994538240674597601587306681545756471103647447e+2561'
+    assert str(result[89]) == '4.02790050127220994538240674597601587306681545756471103647447357787726238637266286878923131618587993e+2561
```

Reply:
@smichr @oscarbenjamin Would like if you could react to the above comments .Thanks !

Reply:
I think it would be fine to check a low precision result like prec=2 and then just a normal precision result (prec=53). Rather than testing the string use `srepr` to show how to create a Float with the correct precision.

Reply:
> Rather than testing the string use `srepr` to show how to create a Float with the correct precision.

I used string to keep this test consistent with other similar tests in the ` test_evalf.py`. I'll try to get familiar with `srepr` and change the tests.